diff --git a/mod.rs b/mod.rs
index 26b9102..6e8c5ae 100644
--- a/mod.rs
+++ b/mod.rs
@@ -249,7 +249,7 @@
 //! time and may call fewer or more syscalls/library functions.
 
 use core::cmp;
-use rustc_unicode::str as core_str;
+use core::str as core_str; //use rustc_unicode::str as core_str;
 use core::fmt;
 use core::iter::{Iterator};
 use core::marker::Sized;
@@ -1635,7 +1635,7 @@ impl<R: Read> Iterator for Chars<R> {
             }
         }
         Some(match str::from_utf8(&buf[..width]).ok() {
-            Some(s) => Ok(s.char_at(0)),
+            Some(s) => s.chars().next().ok_or(CharsError::NotUtf8), //Ok(s.char_at(0)),
             None => Err(CharsError::NotUtf8),
         })
     }
